LiteDB/Engine/Engine/Upgrade.cs

```diff
@@ -19,20 +23,21 @@ private void TryUpgrade()
             // if file not exists, just exit
             if (!File.Exists(filename)) return;
 
+            var buffer = bufferPool.Rent(1024);
             using (var stream = new FileStream(
                 _settings.Filename,
                 FileMode.Open,
                 FileAccess.Read,
                 FileShare.Read, 1024))
             {
-                var buffer = new byte[1024];
+
 
                 stream.Position = 0;
                 stream.Read(buffer, 0, buffer.Length);
```

Comment:
the length is not 1024 anymore! It could be 10 mb depending on what the pool has left.